src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java

```diff
@@ -306,9 +307,18 @@ private void startNextDownloaders() throws SnowflakeSQLException {
 
       // try to reserve the needed memory
       long curMem = currentMemoryUsage.addAndGet(neededChunkMemory);
+      logger.debug(
+          "memoryLimit: {}, curMem: {}, retry: {}, maxRetry: {}",
+          memoryLimit,
+          curMem,
+          retry,
+          MAX_NUM_OF_RETRY);
       // no memory allocate when memory is not enough for prefetch
-      if (curMem > memoryLimit && nextChunkToDownload - nextChunkToConsume > 0) {
+      if (curMem > memoryLimit
```

Comment:
If the amount of memory needed to prefetch the next chunk is greater than the amount of memory available and the very first chunk hasn't even started downloading yet, this likely means we will never have enough memory available to do prefetching at the same time as downloading the current chunk. Cancel the prefetch so as to not block while downloading current chunk, and try again for next chunk.